Improvements to info that is exposed via LSP

wolflu05 opened the pull request:
Hey, thank you really much for developing this LSP. I recently came across it and it turns out to be really useful to use it via [monaco-languageclient](https://github.com/TypeFox/monaco-languageclient) for [inventree](https://github.com/inventree/inventree) to gain autocompletions for admins crafting their custom report/labels.

This PR adds some improvements I came across when trying to get it to work. This is currently a draft as I am still in the process of integrating it and may add some more documentation introspections.

Please let me know if I'm on the right track here and if these changes would be accepted.

krukas commented on djlsp/parser.py:
```diff
@@ -443,6 +450,18 @@ def get_tag_hover(self, line, character, match: Match):
+        context_name = self._get_full_hover_name(line, character, match.group(2))
+        logger.debug(f"Find context hover for: {context_name}")
+
+        docs = list(doc_str for d in self.create_jedi_script(context_name).help() if (doc_str := d.docstring()))
```
I don't like walrus operator in a list comprehension, i rather have extra lines that are easier to read.

krukas reviewed (commented):
Thanks for your contribution! The changes look good to me. I just have one question: How well will a docstring in reStructuredText be rendered when using `MarkupKind.Markdown`?

wolflu05 commented:
Thank you for your feedback. I have changed the walrus operator.

Not sure about the markdown. I was trying to get some better highlighting like vscode does with pyright for python, but comparing it side by side it doesnt look like an improvement. I have removed it now.

If you want, you can already merge this and I create separate PRs for further improvements. I definetly want to look into caching the collected data/provide it via stdin if nothing has changed between lsp server restarts to speed things up and make it more dynamic.

krukas commented on djlsp/parser.py:
```diff
@@ -448,6 +453,22 @@ def get_tag_hover(self, line, character, match: Match):
                 )
         return None
 
+    def get_context_hover(self, line, character, match: Match):
```
Could you add test for this hover, see other for reverence https://github.com/fourdigits/django-template-lsp/blob/main/tests/test_parser.py#L189-L210
